git_wrapper/commit.py

```diff
@@ -0,0 +1,135 @@
+#! /usr/bin/env python
+"""This module acts as an interface for common git tasks"""
+
+import logging
+import os
+
+import git
+
+from git_wrapper.base import GitWrapperBase
+from git_wrapper import exceptions
+from git_wrapper.utils.decorators import reference_exists
+
+
+logger = logging.getLogger(__name__)
+
+
+class GitWrapperCommit(GitWrapperBase):
+    """Provides git commit functionality"""
+
+    def commit(self, message):
+        """Create a commit for changes to tracked files in the repo.
+           Equivalent to `git commit -a -m <message>`.
+
+           :param str message: The commit message
+        """
+        if not message:
+            raise exceptions.CommitMessageMissingException()
```

Comment:
I added a logging statement to this. I left it as debug because I'm a bit wary of having the library spew too much logging as opposed to letting the caller decide... Happy to adjust if that feels off though!